web-services/client/src/main/java/datawave/webservice/model/ModelValidator.java

```diff
@@ -0,0 +1,119 @@
+import java.util.Set;
+
+public class ModelValidator {
+    
+    public static void validate(MetadataHelper metadataHelper, Model model) throws InvalidModelException, TableNotFoundException {
+        
+        Map<String,Set<String>> forwardMappings = buildForwardMappingMap(model);
+        checkFieldProperties(metadataHelper, forwardMappings);
+        // anything else to check?
+        
+    }
+    
+    private static void checkFieldProperties(MetadataHelper metadataHelper, Map<String,Set<String>> forwardMappings) throws TableNotFoundException,
+                    InvalidModelException {
+        
+        StringBuilder messages = new StringBuilder();
+        
+        for (Map.Entry<String,Set<String>> modelField : forwardMappings.entrySet()) {
+            boolean hasIndexedFields = false;
+            boolean hasNonIndexedFields = false;
+            
+            boolean hasReverseIndexedFields = false;
+            boolean hasNonReverseIndexedFields = false;
+            
+            boolean hasTokenizedFields = false;
+            boolean hasNonTokenizedFields = false;
+            
+            Iterator<String> itr = modelField.getValue().iterator();
+            while (itr.hasNext()) {
+                String dbField = itr.next();
+                if (metadataHelper.isIndexed(dbField, UniversalSet.instance())) {
+                    hasIndexedFields = true;
+                } else {
+                    hasNonIndexedFields = true;
+                }
+                
+                if (metadataHelper.isReverseIndexed(dbField, UniversalSet.instance())) {
+                    hasReverseIndexedFields = true;
+                } else {
+                    hasNonReverseIndexedFields = true;
+                }
+                
+                if (metadataHelper.isTokenized(dbField, UniversalSet.instance())) {
+                    hasTokenizedFields = true;
+                } else {
+                    hasNonTokenizedFields = true;
+                }
+                
+            }
+            
+            if (hasIndexedFields && hasNonIndexedFields) {
+                messages.append("Model field ");
+                messages.append(modelField.getKey());
+                messages.append(" has mixed indexed states.");
+                messages.append("\n");
+            }
+            
+            if (hasReverseIndexedFields && hasNonReverseIndexedFields) {
+                messages.append("Model field ");
+                messages.append(modelField.getKey());
+                messages.append(" has mixed reverse index states.");
+                messages.append("\n");
+            }
+            
+            if (hasTokenizedFields && hasNonTokenizedFields) {
+                messages.append("Model field ");
+                messages.append(modelField.getKey());
+                messages.append(" has mixed tokenized states.");
+                messages.append("\n");
+                
+            }
+            
```

Comment:
Is is possible for a field to have reverse indexed stats but not forward indexed status? 

It might be worth checking just in case a field is added as both forward and reverse indexed, but later only the forward indexed state is removed, leaving an "orphaned" reverse indexed state.